src/array.vala

```diff
@@ -44,19 +44,25 @@ public class Vast.Array : Object
 
     public size_t size {
         get {
-            if (dimension == 0) {
-                return 0;
-            } else {
-                size_t size = 1;
-                for (var i = 0; i < dimension; i++) {
-                    size *= _shape[i];
-                }
-                return size;
+            size_t size = 1;
+            for (var i = 0; i < dimension; i++) {
+                size *= _shape[i];
             }
+            return size;
         }
     }
 
-    public Bytes? data { get; construct; default = null; }
+    /* Internal bytes that stores the array; cannot be null */
+    public Bytes? data { get; construct;}
+    internal uint8 * _baseptr;
+
+    construct {
+        /* provide a default buffer is nothing allocated */
+        if(_data == null) {
+            _data = new Bytes (new uint8[scalar_size * size]);
+        } 
+        _baseptr = _data.get_data();
+    }
```

Comment:
This is not good for bindings. I use the `_data == null` for empty array that you can reshape any way you want. This is because we cannot set `shape` from PyGObject.